packages/expo-auth-session/src/__tests__/ManifestTestUtils.ts

```diff
@@ -0,0 +1,50 @@
+import Constants from 'expo-constants';
+
+export type ConstantsType = typeof Constants;
+export type ConstantsWithoutManifests = Omit<ConstantsType, 'manifest' | 'manifest2'>;
+
+export type LegacyManifestPartial = Partial<ConstantsType['manifest']>;
+export type NewManifestPartial = Partial<ConstantsType['manifest2']>;
+
+export type LegacyOrNewManifest =
+  | {
+      manifest: LegacyManifestPartial;
+      manifest2: null;
+    }
+  | {
+      manifest: null;
+      manifest2: NewManifestPartial;
+    };
+
+export const describeManifestTypes =
+  (manifest: LegacyManifestPartial, manifest2: NewManifestPartial) =>
+  (fn: (manifestObj: LegacyOrNewManifest) => any) =>
+    describe.each<[string, LegacyOrNewManifest]>([
+      ['legacy', { manifest, manifest2: null }],
+      ['new', { manifest: null, manifest2 }],
+    ])('manifest type %p', (_, manifestObj) => fn(manifestObj));
+
+/**
+ * Use of this requires use of `require` for the module that imports `expo-constants`
+ * @param constants
+ * @param manifestProvider
+ */
+export function mockConstants(
+  constants: Partial<ConstantsWithoutManifests>,
+  manifestProvider: LegacyOrNewManifest
```

Comment:
nit: IMO this should be called `manifest` or `mockManifest` since a provider implies it's a way to get the manifest rather than the manifest itself